File: algorithms/60.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    string getPermutation(int n, int k)
    {
        vector<int> factorial(10, 1);
        for ( unsigned int i = 2; i < factorial.size(); ++i )
            factorial[i] = factorial[i-1]*i;
        vector<bool> visited(n+1, false);

        k -= 1;
        string result;
        for ( n -= 1; n >= 0; --n )
        {
            int c = k/factorial[n];
            int i;
            for ( i = 1; c >= 0; ++i )
            {
                if ( !visited[i] )
                    c -= 1;
            }
            visited[--i] = true;
            result.push_back('0'+i);
            k = k%factorial[n];
        }
        return result;
    }
};

int main()
{
    Solution solution;
    cout << solution.getPermutation(3, 6) << endl;
    return 0;
}

File: algorithms/python3/60.py
import math

class Solution:
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        k -= 1
        s = [str(i) for i in range(1, n+1)]
        rls = []
        for i in range(n-1, -1, -1):
            fac = math.factorial(i)
            idx = k // fac
            k %= fac
            rls.append(s[idx])
            s = s[:idx] + s[idx+1:]
        return ''.join(rls)
